h2o-algos/src/main/java/hex/rulefit/RuleFitModel.java

```diff
@@ -0,0 +1,139 @@
+
+    public static class RuleFitParameters extends Model.Parameters {
+        public String algoName() {
+            return "RuleFit";
+        }
+
+        public String fullName() {
+            return "RuleFit";
+        }
+
+        public String javaName() {
+            return RuleFitModel.class.getName();
+        }
+
+        @Override
+        public long progressUnits() {
+            return RuleFit.WORK_TOTAL;
+        }
+
+        // the algorithm to use to generate rules. Options are "DRF" (default), "GBM"
+        public Algorithm _algorithm = Algorithm.AUTO;
+
+        // minimum length of rules. Defaults to 1.
+        public int _min_rule_length = 1;
+
+        // maximum length of rules. Defaults to 10.
+        public int _max_rule_length = 10;
+
+        // the maximum number of rules to return. Defaults to -1 which means the number of rules is selected 
+        // by diminishing returns in model deviance.
+        public int _max_num_rules = -1;
+
+        // specifies type of base learners in the ensemble. Options are RULES_AND_LINEAR (initial ensemble includes both rules and linear terms, default), RULES (prediction rules only), LINEAR (linear terms only)
+        public ModelType _model_type = ModelType.RULES_AND_LINEAR;
+    }
+
+    public static class RuleFitOutput extends Model.Output {
+
+        // a set of rules and coefficients
+
+        public double[] _intercept;
+
+        public TwoDimTable _rule_importance = null;
+
+        Key[] treeModelsKeys = null;
+
+        Key glmModelKey = null;
+
+        //  feature interactions ...
+
+        public RuleFitOutput(RuleFit b) {
+            super(b);
+        }
+    }
+
+    public RuleFitModel(Key<RuleFitModel> selfKey, RuleFitParameters parms, RuleFitOutput output, SharedTreeModel[] treeModels, GLMModel glmModel) {
+        super(selfKey, parms, output);
+        this.treeModels = treeModels;
+        this.glmModel = glmModel;
+    }
+
+    @Override
+    public ModelMetrics.MetricBuilder makeMetricBuilder(String[] domain) {
+        assert domain == null;
+        switch (_output.getModelCategory()) {
+            case Binomial:
+                return new ModelMetricsBinomial.MetricBuilderBinomial(domain);
+            case Multinomial:
+                return new ModelMetricsMultinomial.MetricBuilderMultinomial(_output.nclasses(), domain);
+            case Regression:
+                return new ModelMetricsRegression.MetricBuilderRegression();
+            default:
+                throw H2O.unimpl("Invalid ModelCategory " + _output.getModelCategory());
+        }
+    }
+
+    @Override
+    protected double[] score0(double data[], double preds[]) {
+        throw new UnsupportedOperationException("RuleFitModel doesn't support scoring on raw data. Use score() instead.");
+    }
+
+    @Override
+    public Frame score(Frame fr, String destination_key, Job j, boolean computeMetrics, CFuncRef customMetricFunc) throws IllegalArgumentException {
```

Comment:
Should the model itself be represented by rules instead of the original models?

This will be harder to manage - load/save model will not work out of the box and MOJO will be less straightforward. 

this package https://github.com/christophM/rulefit/blob/master/rulefit/rulefit.py does represent the model as rules

cc: @meganjkurka